include/seqan3/range/views/minimiser.hpp

```diff
@@ -0,0 +1,481 @@
+    /*!\brief Returns an iterator to the first element of the range.
+     * \returns Iterator to the first element.
+     *
+     * \details
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    auto begin() noexcept
+    {
+        return window_iterator<urng_t>{std::ranges::begin(urange), std::ranges::end(urange), window_values_size};
+    }
+
+    //!\copydoc begin()
+    auto begin() const noexcept
+    //!\cond
+        requires seqan3::const_iterable_range<urng_t>
+    //!\endcond
+    {
+        return window_iterator<urng_t const>{std::ranges::begin(urange), std::ranges::end(urange), window_values_size};
+    }
+
+    //!\copydoc begin()
+    auto cbegin() const noexcept
+    //!\cond
+        requires seqan3::const_iterable_range<urng_t>
+    //!\endcond
+    {
+        return begin();
+    }
+
+    /*!\brief Returns an iterator to the element following the last element of the range.
+     * \returns Iterator to the end.
+     *
+     * \details
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    auto end() noexcept
+    {
+        return std::ranges::end(urange);
+    }
+
+    //!\copydoc end()
+    auto end() const noexcept
+    //!\cond
+        requires seqan3::const_iterable_range<urng_t>
+    //!\endcond
+    {
+        return std::ranges::end(urange);
+    }
+
+    //!\copydoc end()
+    auto cend() const noexcept
+    //!\cond
+        requires seqan3::const_iterable_range<urng_t>
+    //!\endcond
+    {
+        return end();
+    }
+    //!\}
+};
+
+//!\brief Iterator for calculating minimisers.
+template <std::ranges::view urng_t>
+template <typename rng_t>
+class minimiser_view<urng_t>::window_iterator
+{
+private:
+    //!\brief The iterator type of the underlying range.
+    using it_t = std::ranges::iterator_t<rng_t>;
+    //!\brief The sentinel type of the underlying range.
+    using sentinel_t = std::ranges::sentinel_t<rng_t>;
+
+public:
+    /*!\name Associated types
+     * \{
+     */
+    //!\brief Type for distances between iterators.
+    using difference_type = typename std::iter_difference_t<it_t>;
+    //!\brief Value type of this iterator.
+    using value_type = size_t;
+    //!\brief The pointer type.
+    using pointer = void;
+    //!\brief Reference to `value_type`.
+    using reference = value_type;
+    //!\brief Tag this class as forward iterator.
+    using iterator_category = std::forward_iterator_tag;
+    //!\brief Tag this class depending on which concept `it_t` models.
+    using iterator_concept = std::conditional_t<std::contiguous_iterator<it_t>,
+                                                typename std::forward_iterator_tag,
+                                                seqan3::iterator_tag_t<it_t>>;
```

Comment:
Right now it says "If the range is a continues_ranges this view models a forward_range, otherwise it models whatever the input range models". Which is wrong, since we can only model at most a forward range. So I think this one is correct:

```suggestion
    using iterator_concept = std::conditional_t<std::input_iterator<it_t>,
                                                std::input_iterator_tag,
                                                std::forward_iterator_tag>;
```
I also wonder why this differs from the iterator_category. I know you copied it from the kmer_hash. @eseiler do you remember the rational behind it when you wrote the kmer_hash?
Otherwise, I would make them the same.

Reply:
kmer_hash view:
```cpp
    using iterator_concept = std::conditional_t<std::contiguous_iterator<it_t>,
                                                 typename std::random_access_iterator_tag,
                                                 iterator_tag_t<it_t>>;
```
If the underlying range is contiguous, we can just say we are random access, because we know the memory location of each element.
If not, we need to check the iterator regarding the concept it models / the tag it provides.

So if you model at most forward, just check if the underlying range is forward. Otherwise you are input.

Reply:
Yes, but why is the `iterator_category` different than the  `iterator_concept`?

Reply:
When I remove the `iterator_concept` all test pass as well (and I do not have the issue described below), so I would just remove it?

Reply:
`iterator_category ` is pre-c++20 and is a tag, `iterator_concept` is c++20 and is a concept.
The concepts and tags have different requirements, hence they may be different - maybe not in this view though.

Reply:
Thank you! Do you know if for the respective concept (LegacyInputIterator, input_iterator) there is an actual c++ concept?
I wonder if 
`std::input_iterator` tests for LegacyInputIterator and 
`std::ranges::input_iterator` tests for input_iterator?

Then I would propose
```cpp
    //!\brief Tag this class as forward iterator.
    using iterator_category = std::conditional_t<std::forward_iterator<it_t>,
                                                 std::forward_iterator_tag,
                                                 std::input_iterator_tag>;
    //!\brief Tag this class depending on which concept `it_t` models.
    using iterator_concept = std::conditional_t<std::ranges::forward_iterator<it_t>,
                                                std::forward_iterator_tag,
                                                std::input_iterator_tag>;
```

Reply:
AFAIK the category is something the developer promises, there is just an exposition-only concept (since c++20) for this.

IIRC, in c++20 you do not really need the category because you have concepts.
However (legacy) algorithms may check the category to determine what they can do.
That means you still have to provide the category.
Additionally you can define the concept, which directly refers to the c++20 concepts. This is probably gonna be used somewhere, otherwise it wouldn't be in the standard - maybe for non-c++20 stuff that is aware of c++concepts...

However, both are different - something with proxy iterators....

Couldn't you just set the category to input iff  `seqan3::iterator_tag_t` says input otherwise it's forward?
And the concept would be the same as the category I think

Reply:
> Couldn't you just set the category to input iff seqan3::iterator_tag_t says input otherwise it's forward?

Isn't this equivalent to what I did? If the underlying range is more then input we model forward otherwise input.

> And the concept would be the same as the category I think

After you explanation (thanks) yes, I agree.